ethcore/src/snapshot/consensus/mod.rs

```diff
@@ -51,17 +54,54 @@ pub trait SnapshotComponents: Send {
 		chunk_sink: &mut ChunkSink,
 		preferred_size: usize,
 	) -> Result<(), Error>;
+}
+
+/// A common trait for `BlockChain` (full node) and `HeaderChain` (light client)
+/// used in snapshot restoration.
+pub trait RestorationTargetChain: Send {
+	/// Same as `BlockChain::genesis_hash`.
+	fn genesis_hash(&self) -> H256;
+
+	/// Same as `BlockChain::genesis_header`.
+	fn genesis_header(&self) -> Header;
+
+	/// Same as `BlockChain::block_hash`.
+	fn block_hash(&self, index: BlockNumber) -> Option<H256>;
+
+	/// Same as `BlockChain::block_header_data`.
+	fn block_header_data(&self, hash: &H256) -> Option<Header>;
+
+	/// Same as `BlockChain::add_child`.
+	fn add_child(&self, batch: &mut DBTransaction, block_hash: H256, child_hash: H256);
```

Comment:
If the LightClient doesn't need this – and I don't have an answer to that I'm afraid – perhaps this doesn't need to be in the trait and can be moved back to the `BlockChain` impl?